src/test/validation_chainstatemanager_tests.cpp

```diff
@@ -57,12 +55,13 @@ BOOST_AUTO_TEST_CASE(chainstatemanager)
     // Create a snapshot-based chainstate.
     //
     ENTER_CRITICAL_SECTION(cs_main);
-    CChainState& c2 = manager.InitializeChainstate(GetRandHash());
+    CChainState& c2 = *WITH_LOCK(::cs_main,
+        return &manager.InitializeChainstate(GetRandHash()));
```

Comment:
Doesn't this lock `cs_main` twice ?

Reply:
The test nits are fixed in the first commit of #19604

Reply:
Yep. Brainfart. Just noticed while rebasing https://github.com/bitcoin/bitcoin/pull/19556 and commented without thinking.